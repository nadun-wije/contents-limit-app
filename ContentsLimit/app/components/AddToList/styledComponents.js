import styled from 'styled-components';
import { tablet, colors } from '../../styles/constants';

export const Container = styled.form`
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid black;
  padding: 20px;
  position: fixed;
  bottom: 0;
  width: 100%;
  background-color: ${colors.lightGrey};
  height: 180px;

  @media (min-width: ${tablet}) {
    width: 500px;
    bottom: 10px;
  }
`;

export const Select = styled.select`
  width: 100%;
  border: 1px solid black;
  border-radius: 5px;
  padding: 5px;
  background-color: white;
`;

export const Input = styled.input`
  display: block;
  border: 1px solid black;
  border-radius: 5px;
  padding: 5px;
  margin-bottom: 20px;
  width: 100%;
  background-color: white;
`;

export const Button = styled.button`
  margin-left: 50px;
  border: 3px solid ${colors.green};
  border-radius: 5px;
  padding: 10px 20px;
  font-weight: bold;
  background-color: ${colors.green};
  color: white;

  :hover:enabled {
    background-color: white;
    color: ${colors.green};
  }

  :disabled {
    border-color: grey;
    background-color: grey;
  }
`;
