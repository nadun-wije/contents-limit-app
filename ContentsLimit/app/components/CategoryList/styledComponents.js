import styled from 'styled-components';

export const Container = styled.ul`
  width: 100%;
  margin-left: 30px;
  padding-left: 0;
`;
