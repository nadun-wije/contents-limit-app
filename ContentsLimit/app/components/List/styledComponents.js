/**
 *
 * List Styled Components
 *
 */

import styled from 'styled-components';

export const ListContainer = styled.ul`
  width: 100%;
`;
