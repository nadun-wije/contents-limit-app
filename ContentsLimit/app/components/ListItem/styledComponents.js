import styled from 'styled-components';

export const Container = styled.li`
  width: 100%;
  height: 3em;
  display: flex;
  align-items: center;
  position: relative;
  border-top: 1px solid #eee;

  &:first-child {
    border-top: none;
  }
`;

export const Item = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 100%;
  align-items: center;
`;

export const Price = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 100%;
  align-items: center;
`;
