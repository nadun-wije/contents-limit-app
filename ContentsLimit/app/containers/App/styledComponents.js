import styled from 'styled-components';
import { tablet } from '../../styles/constants';

export const AppContainer = styled.div`
  max-width: 500px;
  margin: 0 auto;
  min-height: 100%;
  font-size: 16px;
  line-height: 20px;

  button {
    cursor: pointer;
  }
`;

export const TotalText = styled.p`
  padding-top: 20px;
  padding-left: 40px;
`;

export const ListContainer = styled.div`
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  width: 100%;
  overflow-x: hidden;
  overflow-y: scroll;

  /* Hide the scrollbar */
  ::-webkit-scrollbar {
    width: 0px;
    background: transparent;
  }

  @media (min-width: ${tablet}) {
    margin-top: 20px;
    width: 500px;
  }
`;
