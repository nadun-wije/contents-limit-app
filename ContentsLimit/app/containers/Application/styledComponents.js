import styled from 'styled-components';

export const AppContainer = styled.div`
  border: 1px solid black;
  max-width: 500px;
  margin: 0 auto;
  display: flex;
  min-height: 100%;
  /* padding: 0 16px; */
  flex-direction: column;
  font-size: 16px;
  line-height: 20px;
  width: 100%;
  overflow-x: hidden;
  overflow-y: scroll;
`;

export const TotalText = styled.p`
  padding-top: 20px;
  padding-left: 40px;
`;
