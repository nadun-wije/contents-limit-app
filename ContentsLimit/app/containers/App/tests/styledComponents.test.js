/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { AppContainer, TotalText, ListContainer } from '../styledComponents';

describe('App styled components', () => {
  describe('AppContainer styled component', () => {
    it('should exist', () => {
      expect(shallow(<AppContainer />).exists()).toBe(true);
    });

    it('should render with styles', () => {
      const tree = renderer.create(<AppContainer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('TotalText styled component', () => {
    it('should exist', () => {
      expect(shallow(<TotalText />).exists()).toBe(true);
    });

    it('should render with styles', () => {
      const tree = renderer.create(<TotalText />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('ListContainer styled component', () => {
    it('should exist', () => {
      expect(shallow(<ListContainer />).exists()).toBe(true);
    });

    it('should render with styles', () => {
      const tree = renderer.create(<ListContainer />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
