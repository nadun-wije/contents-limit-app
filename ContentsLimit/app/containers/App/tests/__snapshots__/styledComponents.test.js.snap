// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App styled components AppContainer styled component should render with styles 1`] = `
.c0 {
  max-width: 500px;
  margin: 0 auto;
  min-height: 100%;
  font-size: 16px;
  font-family: 'Noto Sans',sans-serif;
  color: #4D4D4F;
  line-height: 20px;
}

.c0 button {
  cursor: pointer;
  font-family: 'Noto Sans',sans-serif;
}

.c0 input,
.c0 select {
  font-family: 'Noto Sans',sans-serif;
}

<div
  className="c0"
/>
`;

exports[`App styled components ListContainer styled component should render with styles 1`] = `
.c0 {
  border: 1px solid black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  overflow-x: hidden;
  overflow-y: scroll;
  height: calc(100vh - 180px);
}

.c0::-webkit-scrollbar {
  width: 0px;
  background: transparent;
}

@media (min-width:480px) {
  .c0 {
    margin-top: 20px;
    width: 500px;
  }
}

<div
  className="c0"
/>
`;

exports[`App styled components TotalText styled component should render with styles 1`] = `
.c0 {
  padding-top: 20px;
  padding-left: 40px;
  font-weight: bold;
  color: #5BB947;
}

<p
  className="c0"
/>
`;
